src/m365/pp/commands/managementapp/managementapp-add.ts

```diff
@@ -0,0 +1,130 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+export interface Options extends GlobalOptions {
+  appId?: string;
+  objectId?: string;
+  name?: string;
+}
+
+class PpManagementappAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MANAGEMENTAPP_ADD;
+  }
+
+  public get description(): string {
+    return 'Register management application for Power Platform';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.objectId = typeof args.options.objectId !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getAppObjectId(args)
+      .then((appId: string): Promise<void> => {
+        const requestOptions: any = {
+          // This should be refactored once we implement a PowerPlatform base class as api.bap will differ between envs.
+          url: `https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/adminApplications/${appId}?api-version=2020-06-01`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.put(requestOptions);
+      })
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getAppObjectId(args: CommandArgs): Promise<string> {
+    if (args.options.appId) {
+      return Promise.resolve(args.options.appId);
+    }
+
+    const { objectId, name } = args.options;
+
+    const filter: string = objectId ?
+      `id eq '${encodeURIComponent(objectId)}'` :
+      `displayName eq '${encodeURIComponent(name as string)}'`;
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/myorganization/applications?$filter=${filter}&$select=appId`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: { appId: string }[] }>(requestOptions)
```

Comment:
Let's use the `Application` type from the MS Graph types package instead of defining our own

Reply:
@waldekmastykarz I feel like I need a little more guidance. I had a look at https://github.com/pnp/cli-microsoft365/blob/main/src/m365/aad/commands/app/app-role-delete.ts as we didnt use the the Graph Types there either. Do you want me to refactor to retrieve the whole application (and then use the `Application.AppId` in caller instead of the string?

Reply:
Yes, at some point we decided that we'd move forward with types from the Graph package instead of maintaining our own so when possible, let's try to use them.